components/tools/OmeroWeb/omeroweb/webgateway/views.py

```diff
@@ -1198,6 +1210,40 @@ def wrap(request, *args, **kwargs):
     return wrap
 
 
+def json_response(f):
+    """
+    Decorator for wrapping response in JsonResponse if needed and
+    error handling
+
+    @param f:       The function to wrap
+    @return:        The wrapped function, which will return json
+    """
+    @wraps(f)
+    def wrap(request, *args, **kwargs):
+        logger.debug('json_response')
+        try:
+            rv = f(request, *args, **kwargs)
+            if isinstance(rv, HttpResponse):
```

Comment:
There are several cases where we want to handle exceptions within the Views methods themselves in order to provide a specific error message and response code. The easiest way to do this is simply return the complete HttpResponse.